src/slash-bedrock/share/brl-fetch/distros/exherbo

```diff
@@ -0,0 +1,147 @@
+#!/bedrock/libexec/busybox sh
+#
+# Exherbo Linux bootstrap support
+# Copyright 2019 Wulf C. Krueger <philantrop@exherbo.org>
+# Distributed under the terms of the GNU General Public License v2
+#
+# based in part upon "Gentoo Linux bootstrap support" which is:
+#      This program is free software; you can redistribute it and/or
+#      modify it under the terms of the GNU General Public License
+#      version 2 as published by the Free Software Foundation.
+#
+#	Copyright (c) 2016-2019 Daniel Thau <danthau@bedrocklinux.org>
+
+. /bedrock/share/common-code
+trap 'fetch_abort "Unexpected error occurred."' EXIT
+
+check_supported() {
+	true
+}
+
+speed_test_url() {
+	echo "/stages/sha1sum"
+}
+
+list_mirrors() {
+	echo "https://dev.exherbo.org/"
+}
+
+brl_arch_to_distro() {
+	case "${1}" in
+	     "x86_64") echo "amd64" ;;
+	     *) abort "brl does not know how to translate arch \"${1}\" to ${distro:-} format" ;;
+	esac
+}
+
+list_architectures() {
+	cat <<EOF
+x86_64
+EOF
+}
+
+default_release() {
+	echo "rolling"
+}
+
+list_releases() {
+	echo "rolling"
+}
+
+fetch() {
+        paludis_builddir=/var/paludis
+        paludis_repo_dir=/etc/paludis/repositories
+        paludis_user=paludisbuild
+        paludis_group=paludisbuild
+
+	ex_download=/var/tmp/brl-fetch
+	ex_stage=exherbo-x86_64-pc-linux-gnu-current.tar.xz
+	ex_checksum=sha1sum
+
+	mkdir -p "${ex_download}"
+
+	step "Downloading bootstrap software"
+	(
+	    if [[ -f ${ex_download}/${ex_stage} ]]; then
+	        cp "${ex_download}/${ex_stage}" "${bootstrap_dir:-}"
+		cp "${ex_download}/${ex_checksum}" "${bootstrap_dir:-}"
+	    else
+		wget ${target_mirror:-}/stages/"${ex_stage}" -O "${bootstrap_dir:-}"/"${ex_stage}"
+		wget ${target_mirror:-}/stages/"${ex_checksum}" -O "${bootstrap_dir:-}"/"${ex_checksum}"
+	    fi
+	)
+
+	step "Verifying stage tarball"
```

Comment:
This is great.  I need to add checksumming to other `brl fetch` back ends. :+1:

Reply:
Thanks. Here, too, I think, some global framework would be nice. I know other distros offer the possibility as well.